### PR TITLE
Add new account version that supports MuSig2 v1.0.0-rc2 protocol (coming with lnd 0.16.0-beta)

### PR DESCRIPTION
**Pull request by guggero**
~Depends on https://github.com/lightningnetwork/lnd/pull/7171.~
~Depends on https://github.com/lightninglabs/lndclient/pull/139.~

This PR add support for the updated MuSig2 `v1.0.0-rc2` protocol to Pool accounts.

The order for getting this in is:
- [x] Merge https://github.com/lightningnetwork/lnd/pull/7171
- [x] Release `lnd v0.16.0-beta(.rcX)`
- [x] Merge https://github.com/lightninglabs/lndclient/pull/139
- [x] Merge the Faraday PR (https://github.com/lightninglabs/faraday/pull/157)
- [x] Merge the Aperture PR (https://github.com/lightninglabs/aperture/pull/85)
- [x] Merge the server side PR and deploy it to mainnet
- [ ] Merge this PR

#### Pull Request Checklist
- [X] LndServices minimum version has been updated if new lnd apis/fields are
  used.

### CODE REVIEW

**Comment by guggero** [comment]
@positiveblue Christmas came early this year, we now have feature flag bases batch versions :sunglasses:

**Review by positiveblue — APPROVED** [review]
:tada:

File: rpcserver.go [hunk 1]
```diff
@@ -3026,19 +3029,49 @@ func (s *rpcServer) setTicketStateForOrder(newState sidecar.State,
...
 		return account.VersionTaprootEnabled, nil
-	}
 
-	return account.VersionTaprootEnabled, nil
+	case poolrpc.AccountVersion_ACCOUNT_VERSION_TAPROOT_V2:
+		if !isMuSig2V100RC2Compatible {
+			return 0, fmt.Errorf("cannot use Taproot v2 enabled "+
+				"account version, the lnd node Pool is "+
+				"connected to is version %v but needs to be "+
+				"at least %v", currentVersion.AppMinor,
```
Comment (ffranr) [inline rpcserver.go:3066]: nit, consider: 

```
account version does not support Taproot v2: pool is connected to lnd node version %v but requires at least %v
```

File: account/manager_test.go [hunk 2]
```diff
@@ -1152,6 +1221,65 @@ func TestAccountDeposit(t *testing.T) {
 			version:    VersionInitialNoVersion,
 			newVersion: VersionTaprootEnabled,
 		},
+		{
+			name: "happy path version 2",
+			fundedOutputAmount: depositAmount +
+				accountInputFeesTaproot,
+
+			// We'll use the lowest fee rate possible, which should
+			// yield a transaction fee of 304 satoshis when taking
```
Comment (ffranr) [inline account/manager_test.go:1230]: nit: I think that specifics like "304" will become stale as they are hard to keep up-to-date. A assertion in code would be an alternative. I would suggest removal from the comment.

File: order/batch.go [hunk 3]
```diff
@@ -76,37 +78,68 @@ const (
...
 func (bv BatchVersion) SupportsAccountTaprootUpgrade() bool {
-	return bv >= UpgradeAccountTaprootBatchVersion
+	return (bv & LinearVersionEnd) >= UpgradeAccountTaprootBatchVersion
+}
+
+// SupportsAccountTaprootV2Upgrade is a helper function to easily check if a
+// version supports upgrading SegWit v0 (p2wsh) or Taproot v2 (p2tr) to
+// Taproot v2 (p2tr) or not.
+func (bv BatchVersion) SupportsAccountTaprootV2Upgrade() bool {
+	return bv&UpgradeAccountTaprootV2Flag == UpgradeAccountTaprootV2Flag
```
Comment (ffranr) [inline order/batch.go:129]: nit: match formatting as above, therefore:
```golang
return (bv & UpgradeAccountTaprootV2Flag) == UpgradeAccountTaprootV2Flag
```

**Review by ffranr — APPROVED** [review]
Nice testing!

File: account/manager_test.go (hunk 2 above)
Comment (guggero) [reply account/manager_test.go:1228]: Yeah, good call. I replaced the magic number with a constant and removed the comments.

**Pull request merged**